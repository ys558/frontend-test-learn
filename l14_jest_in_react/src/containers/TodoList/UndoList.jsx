import React, { Component } from 'react'

export default class UndoList extends Component {
  constructor(props){
    super(props)
  }
  render() {
    const { undoList, removeItem } = this.props
    return (
      <div>
        <div data-test="count">{undoList.length}</div>
        <ul>
        {undoList.map((item, index) => 
          <li key={`${item}-${index}`} data-test="list-item"> {item}
            <button onClick={()=>removeItem(index)} data-test="remove-item">
              remove
            </button>
          </li>
        )}
        </ul>
      </div>
    )
  }
}
