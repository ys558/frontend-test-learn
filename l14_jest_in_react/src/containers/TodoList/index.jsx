import React, { useState } from 'react';

const TodoList = () => {
  const [undoItem, addUndoItem] = useState([])
  return <div>
    <Input onAddUndoItem={value => addUndoItem([value, ...undoItem]) }  />
    <UndoList undoItem={undoItem} addUndoItem={addUndoItem} />
  </div>
}

const Input = ({ onAddUndoItem }) => {
  const [inputVal, setInputVal] = useState('')
  const onChange = e => setInputVal(e.target.value)
  const onKeyUp = e => { 
    if (e.keyCode === 13 && inputVal !=='' ) {
      onAddUndoItem(inputVal)
      setInputVal('')
    }
    return 
  }
  return <div>
    <div>Todo List:</div>
    <input type="text" value={inputVal} onChange={onChange} onKeyUp={onKeyUp} dbbtest="inputer" />
  </div>
}

const UndoList = ({undoItem, addUndoItem }) => {
  const delFruit = index => {
    const temp = [...undoItem]
    temp.splice(index, 1)
    addUndoItem(temp)
  }
  return <div>
    {undoItem.map((item, index)=> <div key={index}>
      <span>{item}{' '}
        <button onClick={()=> delFruit(index)}>Ã—</button>
      </span>
    </div>)}
  </div>
}

export default TodoList