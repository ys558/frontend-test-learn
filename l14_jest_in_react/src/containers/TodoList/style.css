* {
  margin: 0;
  padding: 0;
}
.header {
  line-height: 60px;
  background-color: #333;
}

.header-content {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 24px;
  color: #fff;
  width: 600px;
  margin: 0 auto;
}

.header-input {
  /* 去除蓝色边框: */
  outline: none;
  width: 360px;
  float: right;
  margin-top: 15px;
  line-height: 24px;
  border-radius: 5px;
  padding: 0 10px;
}