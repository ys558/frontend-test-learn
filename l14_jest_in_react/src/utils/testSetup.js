// 测试共用配置: 
// 1. 再去 package.json里修改
// "setupFilesAfterEnv": [
//   "<rootDir>/src/utils/testSetup.js"
// ],
// 2. react-create-app v3.3+, yarn eject弹出后, 有直接的配置目录:写在这里也行:
// \src\setupTests.js
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });