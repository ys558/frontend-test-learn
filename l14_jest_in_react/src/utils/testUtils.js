// 封装findTestWrapper方法, 用于选择到component里对应的组件:
export const findTestWrapper = (wrapper, tag) => wrapper.find(`[bddtest='${tag}']`)