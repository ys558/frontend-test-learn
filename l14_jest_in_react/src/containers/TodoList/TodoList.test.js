import React from 'react'
import { mount, shallow } from 'enzyme'
import TodoList from '../TodoList/index'
import { findTestWrapper } from '../../../src/utils/testUtils'
import { render, cleanup, debug } from '@testing-library/react'

describe('输入框测试', ()=> {
  let wrapper
  beforeEach(()=> wrapper= mount(<TodoList/>))
  
  test('1. Input输入框输入内容; 2. 点击回车; 3. 列表中展示用户输入的内容', ()=>{
    const wrapper = mount(<TodoList/>)
    expect()
  })
})
