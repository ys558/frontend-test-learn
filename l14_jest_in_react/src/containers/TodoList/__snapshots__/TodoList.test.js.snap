// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. Input输入框输入内容; 2. 点击回车; 3. 列表中展示用户输入的内容 1`] = `
<TodoList>
  <div>
    <Input
      onAddUndoItem={[Function]}
    >
      <div>
        <div>
          Todo List:
        </div>
        <input
          bddtest="inputer"
          onChange={[Function]}
          onKeyUp={[Function]}
          type="text"
          value=""
        />
      </div>
    </Input>
    <UndoList
      addUndoItem={[Function]}
      undoItem={Array []}
    >
      <div />
    </UndoList>
  </div>
</TodoList>
`;
